### PR TITLE
Add a parser worker to the project.

### PR DESCRIPTION
**Pull request by wldcordeiro**
Associated Issue: #2282 

### Summary of Changes

* refactor parser to introduce a worker

### CODE REVIEW

File: src/components/Editor/SearchBar.js [hunk 1]
```diff
@@ -262,11 +264,9 @@ const SearchBar = React.createClass({
       return;
     }
 
-    const symbolDeclarations = getSymbols(
-      sourceText.toJS())[selectedSymbolType];
-
+    const symbolDeclarations = await getSymbols(sourceText.toJS());
```
Comment (wldcordeiro) [inline src/components/Editor/SearchBar.js]: Turns out trying to access a member of something you're awaiting *immediately* can cause undefined returns. Moving the access to afterwards fixed the issue of `getSymbols` returning  undefined.

**Comment by wldcordeiro** [comment]
mochitest is failing because it can't find the parser-worker.js file? 🙁

File: src/components/Editor/index.js [hunk 2]
```diff
@@ -292,11 +293,11 @@ const Editor = React.createClass({
     }
 
     const variables = selectedFrame.scope.bindings.variables;
-    const expression = getExpressionFromToken(cm, sourceText, token);
+    const expression = await getExpressionFromToken(cm, sourceText, token);
 
     if (variables.hasOwnProperty(tokenText) || expression ||
       tokenText == "this" && selectedFrame.this) {
-      this.setState({ selectedToken: token });
+      this.setState({ selectedToken: token, expression });
```
Comment (jasonLaster) [inline src/components/Editor/index.js]: nit: how about `selectedExpression`?

Comment (wldcordeiro) [inline src/components/Editor/index.js]: Yeah that's a better name. 👍

File: src/components/Editor/SearchBar.js (hunk 1 above)
Comment (jasonLaster) [inline src/components/Editor/SearchBar.js]: oh nice - that makes sene

**Pull request merged**